### ISSUE TITLE
FlxGroup add to end of members

### ISSUE THREAD

**Issue by NQNStudios** [HaxeFlixel/flixel#2613]
Currently, `FlxGroup.add()` has no option to guarantee that the new object will be added at the end of its member array. If I want to `remove()` and `add()` an object to bring it visually to the front of the game, I can only do that if I remember to call `remove()` with the `splice` argument true every time. Additionally, if I use any library/plugin that calls `remove()` without splice, I open myself up to Z-sorting errors.

I could manually add the element to `members`, but this won't update the FlxGroup's length. 

I suggest that `add()` could take an optional "at end" argument, which guarantees that the given object is `push()`ed onto members, not put at the first null index.

**Comment by NQNStudios** [comment]
I think the best available workaround at this point is to manually splice `members` to remove every null index before calling `add()`.

**Comment by Geokureli** [comment]
Agreed, I think you should splice members when removing to ensure that there are no null indices in the group. this is what I do. There are also a lot of techniques to use when z-order is important, like using [FlxSort](https://api.haxeflixel.com/flixel/util/FlxSort.html) on groups, or having foreground, midground, and background groups to ensure different objects always appear on top or bottom of each other.
